src/main/scala/dsls/regex/RegularExpression.scala

```diff
@@ -34,3 +59,27 @@ case class Concat(val left: RegularExpression, val right: RegularExpression)
  *  expression
  */
 case class Star(val expression: RegularExpression) extends RegularExpression
+
+object RegularExpressionImplicits {
+   implicit def CharToLiteral(value : Char) =
+                                    new Literal(value)
+   implicit def StringToConcat(value : String): Concat =
+   {
+     val literals : Seq[Literal] = value.map { x => Literal(x) }
+     val result = literals.foldLeft(null : Concat) 
+     { (z : Concat, i : RegularExpression) =>
+       if (z != null)
+         // we don't want null's in our concat's so we will check if left or
+         // right is null and set either to i instead of null if it is
+         if (z.left == null)
+           Concat(z.right, i)
+         else if (z.right == null)
+           Concat(z.left, i)
+         else
+           Concat(z,i)
+       else
+         Concat(z,i)
+     }
+     result
+   }
```

Comment:
Nice use of higher order functions.  I think if you used your start character as `EPSILON` instead of `null` then you wouldn't have to worry about special casing for null.